### ISSUE TITLE
Cilium-Agent PostStartScript fails

### ISSUE THREAD

**Issue by hans-fischer** [cilium/cilium#31471]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

I try to setup a new Kubernetes Cluster and install Cilium after a fresh Ubuntu 22.04-Server (Cloud-init) installation. The Setup is On-Premise at my home.

When Cilium is starting, the cilium-agent-pods report an error in the PostStartScript.

As a kind of debug I tried to modify the DaemonSet of the Agent and removed the postStartScript, after that, Cilium starts like expected without any errors.

What could cause the PostStartScript to fail?

### Cilium Version

I install with Helm (all defaults but debug set to true) in Version 1.15.2

### Kernel Version

Ubuntu 22.04.4 LTS (GNU/Linux 5.15.0-101-generic x86_64)

### Kubernetes Version

v1.29.2

### Sysdump

Without any modification (fails)
[cilium-sysdump-20240319-090440.zip](https://github.com/cilium/cilium/files/14647420/cilium-sysdump-20240319-090440.zip)

After removing the PostStartScript
[cilium-sysdump-20240319-090926.zip](https://github.com/cilium/cilium/files/14647474/cilium-sysdump-20240319-090926.zip)

### Relevant log output

```shell
When the PostStartScript fails, there is no output in kubectl logs and nothing to provide in /var/log/comntainer :(
```

### Anything else?

Thank you for your help. I hope we can get this problem out of my cilium journey.

### Cilium Users Document

- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by squeed** [comment]
AIUI the only way to get the actual output of the failed PostStart hook is via the kubelet logs, via journalctl on the node. Can you retrieve that?

**Comment by hans-fischer** [comment]
@squeed There is nothing sus in the kubelet-logs so far. Or maybe I dont see it.

[kubelet_logs.txt](https://github.com/cilium/cilium/files/14869546/kubelet_logs.txt)

**Comment by lmb** [comment]
Seems like those logs only cover ~less than a minute and they don't mention anything cilium specific at all. What steps did you follow to produce the output?

Maybe something you can try: redirect the output to a file and grab that? https://kubernetes.io/docs/tasks/configure-pod-container/attach-handler-lifecycle-event/#define-poststart-and-prestop-handlers

**Comment by hans-fischer** [comment]
Short Update on this Issue:

I used to run my servers with Ubuntu 22.04, and this version only has Conmon 1.x.x in its repositories. This Conmon version is a little old and has an incompatibility with recent Kubelet versions.

The effect was that every liveness, readiness, or startup probe of type exec 'failed.' This took a long time for me to discover. I hope it helps someone out there.

After a distro update to 24.04 and a Conmon upgrade to 2.x.x, the problem was resolved.
